components/web_discovery/browser/double_fetcher.cc

```diff
@@ -0,0 +1,132 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/web_discovery/browser/double_fetcher.h"
+
+#include <utility>
+
+#include "brave/components/web_discovery/browser/pref_names.h"
+#include "brave/components/web_discovery/browser/request_queue.h"
+#include "brave/components/web_discovery/browser/util.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/mojom/url_response_head.mojom.h"
+
+namespace web_discovery {
+
+namespace {
+constexpr char kUrlKey[] = "url";
+constexpr char kAssociatedDataKey[] = "assoc_data";
+
+constexpr base::TimeDelta kRequestMaxAge = base::Hours(1);
+constexpr base::TimeDelta kMinRequestInterval =
+    base::Minutes(1) - base::Seconds(5);
+constexpr base::TimeDelta kMaxRequestInterval =
+    base::Minutes(1) + base::Seconds(5);
+constexpr size_t kMaxRetries = 3;
+constexpr size_t kMaxDoubleFetchResponseSize = 2 * 1024 * 1024;
+
+constexpr net::NetworkTrafficAnnotationTag kFetchNetworkTrafficAnnotation =
+    net::DefineNetworkTrafficAnnotation("wdp_doublefetch", R"(
+    semantics {
+      sender: "Brave Web Discovery Double Fetch"
+      description:
+        "Retrieves a page of interest without session cookies for
```

Comment:
I would avoid using the word "session" here because we drop all cookies, not just _session cookies_.
```suggestion
        "Retrieves a page of interest without cookies for
```

Reply:
072fdfc3176b60fd71cedf9c05ed83a3420e7415